### PR TITLE
Ldap mailing lists

### PR DESCRIPTION
**Pull request by chibenwa**
Feel free to continue it!

### CODE REVIEW

**Comment by chibenwa** [comment]
Remains:

 - [ ] Antora doc
 - [ ] Rebase https://github.com/apache/james-project/pull/2359
 - [ ] Add mailing lists into the example docker-compose

**Review by Arsnael — COMMENTED** [review]
Read it nothing to say so far

**Comment by chibenwa** [comment]
Results of DEMO tests

Screenshot from 2024-07-24 00-01-52

Screenshot from 2024-07-24 00-18-43

**Comment by chibenwa** [comment]
More or less complete with the Antora doc ;-)

File: demo/docker-compose.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ services:
       - tmail
 
   tmail-backend:
-    image: linagora/tmail-backend:memory-branch-master
+    image: chibenwa/tmail-backend:23072024-1
```
Comment (Arsnael) [inline demo/docker-compose.yml:20]: I understand it's likely for CI tests, but dont forget to put it back to its normal value

File: demo/ldap/populate.ldif [hunk 2]
```diff
@@ -38,4 +38,18 @@ givenName: Twake Mail
...
+
+dn: cn=group, ou=lists, dc=tmail,dc=com
+objectclass: groupofnames
+businessCategory: openList
+cn: group
+ou: lists
+description: group@lists.tmail.com
+member: uid=james-user@tmail.com,ou=users,dc=tmail,dc=com
+member: uid=james-user2@tmail.com,ou=users,dc=tmail,dc=com
+>>>>>>> 7a647dfe (ISSUE-1134 Integrate mailing lists with demo setup)
```
Comment (Arsnael) [inline demo/ldap/populate.ldif:55]: oups?

File: docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+This can be done through the `LDAPMailingList` mailet.
+
+The list address is held by a configurable LDAP attribute of the LDAP group representing the list, and is resolved to
+the list of members of the group.
+
+Twake mail define some sender controls, defined by the business category of the LDAP group:
+
+ - *openList(default) every body can send mails to the list
+ - *internalListonly james users can send mail to the list
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc:14]:
```suggestion
 - *internalList only james users can send mail to the list
```

File: docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+ - *memberRestrictedListonly members of the list can send mail to the list
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc:15]:
```suggestion
 - *memberRestrictedList only members of the list can send mail to the list
```

File: docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+ - *ownerRestrictedListonly owners of the list can send mail to the list
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc:16]:
```suggestion
 - *ownerRestrictedList only owners of the list can send mail to the list
```

File: docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+ - *domainRestrictedListonly members of the domain can send mail to the list
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/features/ldapMailingLists.adoc:17]:
```suggestion
 - *domainRestrictedList only members of the domain can send mail to the list
```

File: demo/docker-compose.yml (hunk 1 above)
Comment (chibenwa) [inline demo/docker-compose.yml:20]:
```suggestion
    image: linagora/tmail-backend:memory-branch-master
```

File: demo/ldap/populate.ldif (hunk 2 above)
Comment (chibenwa) [inline demo/ldap/populate.ldif:55]:
```suggestion
```

**Comment by chibenwa** [comment]
Thanks ;-)

Obviously working late leads to typos ^^

**Review by Arsnael — APPROVED** [review]

**Comment by quantranhong1999** [comment]
read it (late though)

**Pull request merged**